### ISSUE TITLE
Unclean exception when trying to set grouped roles to a user with a non-grouped role

### ISSUE THREAD

**Issue by mbittmann** [alanning/meteor-roles#233]
I think I may have found a bug in error handling. 

I had a user with a role that was not associated with any groups. 
`Roles.setUserRoles(Meteor.userId(), ['some-role']);`

Then after that I tried to add that user to grouped roles:
`Roles.setUserRoles(Meteor.userId(), ['other1', 'other2'], group_name)`

Both were set server side. I would have expected to get an error `...Can't mix grouped and non-grouped ...` but instead got:

```
I20170224-10:33:55.582(-5)? Exception in onLogin callback: TypeError: Cannot read property 'indexOf' of undefined
I20170224-10:33:55.582(-5)?     at strContains (packages/alanning_roles.js:789:25)
I20170224-10:33:55.582(-5)?     at packages/alanning_roles.js:784:12
I20170224-10:33:55.582(-5)?     at Array.some (native)
I20170224-10:33:55.583(-5)?     at Function._.some._.any (packages/underscore.js:267:59)
I20170224-10:33:55.583(-5)?     at isMongoMixError (packages/alanning_roles.js:783:12)
I20170224-10:33:55.583(-5)?     at Object._.extend._updateUserRoles (packages/alanning_roles.js:762:39)
I20170224-10:33:55.583(-5)?     at Object._.extend.addUsersToRoles (packages/alanning_roles.js:221:11)
I20170224-10:33:55.583(-5)?     at app/app.js:2064:17
```

**Comment by alanning** [comment]
Hi Mark, thanks for reporting this.  I have noticed this happen when
MongoDB changes their implementation.  Which DB version did you see this
with?

Adrian

**Comment by mbittmann** [comment]
I'm using the embedded mongo 3.2.6 with Meteor 1.4.2.3

**Comment by mitar** [comment]
I think they again changed the exception format?

**Comment by alanning** [comment]
Yes, that's usually the cause when this breaks.  Strangely enough the unit
test for this case is passing with latest Meteor (METEOR@1.4.2.7).

Mark, could you try running the roles unit tests and see if the "Mixing
groups and non-groups" test fails, please?  Not quite sure how to get it to
use your version of meteor.  Perhaps you can clone the meteor-roles repo
into your packages dir and run the tests that way.

```
meteor test-packages ./roles
```

**Comment by mbittmann** [comment]
Yep, it passes, and it appears to be running the same version of meteor. 
`PASS 73 ms S: mixing group with non-group throws descriptive error`

```
cd .meteor/local/build/programs/server/packages/
git clone https://github.com/alanning/meteor-roles.git
cd meteor-roles/
meteor test-packages ./roles
```

Not sure if it matters, but meteor is telling me `Your development database is using mmapv1, the old, pre-MongoDB 3.0 database engine. You should consider upgrading to Wired Tiger, the new engine.`

**Comment by alanning** [comment]
Hmm, I'd like to have the code support the older versions too.  With the code in your packages directory you should be able to console log the actual error that is being thrown.  Could you re-run your in-app test and send the results to us please?  

Even better if you'd like to do a pull request for handling whatever error message you are seeing from MongoDB!  :-)

**Comment by mbittmann** [comment]
Weird. If I `meteor remove alanning:roles` and instead clone it into a top level packages directory, then I get the appropriate error messages. In both cases the version is 1.2.15. I'm totally new to using custom packages, so it is very possible I'm doing something wrong. I have so far only used `meteor add` for third party packages.

**Comment by mitar** [comment]
I think the issue is that some recent fixes have not yet been released. Like this one: https://github.com/alanning/meteor-roles/commit/eb3376ae0c288a0d6c3b8acbfc2730dc722a3dd0

@alanning: maybe we should make a release for 1x version?

**Comment by alanning** [comment]
Sounds good, would you like to do the honors? :-)

**Comment by mbittmann** [comment]
I can confirm that my `.meteor/local/build/programs/server/packages/alanning_roles.js` when using `meteor add` is:

`if (ex.name === 'MongoError' && isMongoMixError(ex.err)) {`

**Comment by mitar** [comment]
Done. Released 1.2.16.

**Comment by mbittmann** [comment]
You guys are awesome. I just updated the package and can confirm that I'm getting the correct message:

```
I20170228-17:05:41.680(-5)? Exception in onLogin callback: Error: Roles error: Can't mix grouped and non-grouped roles for same user
I20170228-17:05:41.680(-5)?     at Object._.extend._updateUserRoles (packages/alanning_roles.js:763:15)
I20170228-17:05:41.680(-5)?     at Object._.extend.setUserRoles (packages/alanning_roles.js:250:11)
I20170228-17:05:41.681(-5)?     at app/app.js:2058:19
I20170228-17:05:41.681(-5)?     at runAndHandleExceptions (packages/callback-hook.js:152:24)
I20170228-17:05:41.681(-5)?     at packages/callback-hook.js:159:12
```
